qiskit/algorithms/optimizers/l_bfgs_b.py

```diff
@@ -12,11 +12,17 @@
 
 """Limited-memory BFGS Bound optimizer."""
 
-from scipy import optimize as sciopt
-from .optimizer import Optimizer, OptimizerSupportLevel
+from typing import Optional
 
+import numpy as np
 
-class L_BFGS_B(Optimizer):  # pylint: disable=invalid-name
+from qiskit.utils.deprecation import deprecate_arguments
+
+from .optimizer import Optimizer
+from .scipy_minimizer import ScipyMinimizer
+
+
+class L_BFGS_B(ScipyMinimizer):  # pylint: disable=invalid-name
```

Comment:
Is `L_BFGS_B` the only optimizer where we have problems with the return type of the gradients?

Reply:
CG works with gradient. I will check others later.

Reply:
I have checked and fixed it in #6373.